### PR TITLE
Chore: (Docs) Adds test coverage to the docs

### PR DESCRIPTION
**Pull request by jonniebigodes**
This pull request updates the documentation to include test coverage with the associated addon.

What was done:
- Updated the toc
- Updated all the relevant sections to feature in the new docs
- Created the required snippets / assets
- Wrote the docs for coverage tests.

### CODE REVIEW

File: docs/snippets/common/storybook-coverage-addon-install.npm.js.mdx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```shell
+npm install nyc @storybook/addon-coverage --save-dev
```
Comment (yannbf) [inline docs/snippets/common/storybook-coverage-addon-install.npm.js.mdx:2]: nyc is a dependency of the test-runner so users will already have it in their projects

File: docs/snippets/common/storybook-coverage-addon-registration.ts.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+```ts
+// .storybook/main.ts
+
+// Imports Storybook's configuration API
+import type { StorybookConfig } from '@storybook/types';
```
Comment (yannbf) [inline docs/snippets/common/storybook-coverage-addon-registration.ts.mdx:5]: Is this the correct way to import config types @kasperpeulen ? I thought it'd be coming from their framework e.g.

```ts
import type { StorybookConfig } from '@storybook/react-webpack5';
````

File: docs/writing-tests/test-coverage.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+![Test coverage](./test-coverage-storybook.gif)
+
+## Code instrumentation with the coverage addon
+
+Storybook provides an official [test coverage addon](https://storybook.js.org/addons/@storybook/addon-coverage). Powered by [Instanbul](https://istanbul.js.org/), which allows out-of-the-box code instrumentation for the most commonly used frameworks and builders in the JavaScript ecosystem.
+
+### Set up the coverage addon
+
+To enable test coverage with Storybook, you'll need to take additional steps to set it up properly. We recommend you go through the [test runner documentation](./test-runner.md) before proceeding with the rest of the required configuration.
```
Comment (yannbf) [inline docs/writing-tests/test-coverage.md:17]: Maybe make it more explicitly that users should use the test-runner in order to get optimal experience? The addon does help if users want to use something like playwright directly, but it's best suited for test-runner users

File: docs/writing-tests/test-coverage.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+<!-- prettier-ignore-end -->
+
+---
+
+## Troubleshooting
+
+### Run test coverage in other frameworks
+
+If you intend on running coverage tests in other frameworks (e.g., Vue, Svelte), you'll need to adjust your configuration and enable the required file extensions. For example, if you're using Vue, you'll need to add the following to your nyc configuration file (i.e., `nycrc.json` or `nyc.config.js`):
```
Comment (yannbf) [inline docs/writing-tests/test-coverage.md:125]:
```suggestion
If you intend on running coverage tests in frameworks with special files like Vue or Svelte, you'll need to adjust your configuration and enable the required file extensions. For example, if you're using Vue, you'll need to add the following to your nyc configuration file (i.e., `nycrc.json` or `nyc.config.js`):
```

**Review by yannbf — APPROVED** [review]

File: docs/snippets/common/storybook-coverage-addon-registration.ts.mdx (hunk 2 above)
Comment (jonniebigodes) [reply docs/snippets/common/storybook-coverage-addon-registration.ts.mdx:5]: @yannbf both work, but better be safe than sorry. I'll update the snippet to feature a more agnostic approach.

File: docs/snippets/common/storybook-coverage-addon-install.npm.js.mdx (hunk 1 above)
Comment (jonniebigodes) [reply docs/snippets/common/storybook-coverage-addon-install.npm.js.mdx]: Fixed with the latest commit.

File: docs/snippets/common/storybook-coverage-addon-registration.ts.mdx (hunk 2 above)
Comment (jonniebigodes) [reply docs/snippets/common/storybook-coverage-addon-registration.ts.mdx]: Fixed with the latest commit

**Pull request merged**